plugins/modules/system/parted.py

```diff
@@ -500,25 +500,22 @@ def check_parted_label(device):
     return False
 
 
-def parted_version():
+def parted_fetch_version(out):
     """
-    Returns the major and minor version of parted installed on the system.
+    returns version string from the output of "parted --version" command
     """
-    global module, parted_exec
-
-    rc, out, err = module.run_command("%s --version" % parted_exec)
-    if rc != 0:
-        module.fail_json(
-            msg="Failed to get parted version.", rc=rc, out=out, err=err
-        )
-
     lines = [x for x in out.split('\n') if x.strip() != '']
     if len(lines) == 0:
-        module.fail_json(msg="Failed to get parted version.", rc=0, out=out)
+        return
```

Comment:
```suggestion
        return None, None, None
```
(same for the other `return` statements in this function).

Otherwise, you'll get a traceback before `module.fail_json` is called by the caller.